server/modlog.ts

```diff
@@ -74,19 +74,21 @@ export interface ModlogSearch {
 
 export interface ModlogEntry {
 	action: string;
-	roomID?: string;
-	visualRoomID?: string;
-	userid?: ID;
-	autoconfirmedID?: ID;
-	alts?: ID[];
-	ip?: string;
-	isGlobal?: boolean;
-	loggedBy?: ID;
-	note?: string;
+	roomID: string;
+	visualRoomID: string;
+	userid: ID | null;
+	autoconfirmedID: ID | null;
+	alts: ID[];
+	ip: string;
```

Comment:
Is `ip` always present? I see you `|| ''` elsewhere. If not, I would choose a strategy consistent with `userid`--empty string isn't a valid IP address.

Reply:
no, it's not, i'll make it null